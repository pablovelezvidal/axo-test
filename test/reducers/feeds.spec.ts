import { AnyAction } from 'redux';
import feedsReducer, {
  getFeedsFailure,
  getFeeds,
  getFeedsSuccess,
} from '../../app/slices/feedsSlice';

describe('reducers', () => {
  describe('feeds', () => {
    it('should handle initial state', () => {
      expect(feedsReducer(undefined, {} as AnyAction)).toMatchSnapshot();
    });

    it('should handle getFeedsFailure', () => {
      expect(
        feedsReducer(undefined, { type: getFeedsFailure })
      ).toMatchSnapshot();
    });

    it('should handle getFeeds', () => {
      expect(feedsReducer(undefined, { type: getFeeds })).toMatchSnapshot();
    });

    it('should handle getFeedsSuccess', () => {
      expect(
        feedsReducer(
          {
            loading: false,
            feeds: [],
            hasErrors: false,
          },
          {
            type: getFeedsSuccess,
            payload: [
              { id: 1, text: '1' },
              { id: 2, text: '2' },
            ],
          }
        )
      ).toMatchSnapshot();
    });

    it('should handle unknown action type', () => {
      expect(
        feedsReducer(
          { loading: true, feeds: [], hasErrors: false },
          { type: 'unknown' }
        )
      ).toMatchSnapshot();
    });
  });
});
