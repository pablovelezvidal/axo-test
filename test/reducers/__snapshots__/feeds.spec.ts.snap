// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reducers feeds should handle getFeeds 1`] = `
Object {
  "feeds": Array [],
  "hasErrors": false,
  "loading": true,
}
`;

exports[`reducers feeds should handle getFeedsFailure 1`] = `
Object {
  "feeds": Array [],
  "hasErrors": true,
  "loading": false,
}
`;

exports[`reducers feeds should handle getFeedsSuccess 1`] = `
Object {
  "feeds": Array [
    Object {
      "id": 1,
      "text": "1",
    },
    Object {
      "id": 2,
      "text": "2",
    },
  ],
  "hasErrors": false,
  "loading": false,
}
`;

exports[`reducers feeds should handle initial state 1`] = `
Object {
  "feeds": Array [],
  "hasErrors": false,
  "loading": false,
}
`;

exports[`reducers feeds should handle unknown action type 1`] = `
Object {
  "feeds": Array [],
  "hasErrors": false,
  "loading": true,
}
`;
